### PR TITLE
Update model.py to detach diff_token_repr

### PR DESCRIPTION
**Pull request by simonlevine**
I believe `diff_token_repr` is supposed to be detached. I don't think this impacts the way training was set up (serially between trunk and confidence), as relevant parameter gradients are disabled, nor would it impact inference. However, I believe concurrent training would cause an issue as we would backprop from the confidence head to the diffusion module and trunk.

### CODE REVIEW

**Comment by gcorso** [comment]
Hi @simonlevine, 

You are correct that the single representation should be detached! However, the detachment should not happen where you placed it, but right after it comes out of the denoising model as there are a few weights in the recursive inclusion of the token features (OutTokenFeatUpdate) that should still be updated based on the gradients. See my latest commit!

Let me know if this makes sense and thanks a lot anyway for pointing this out!
Gabriele
